### DISCUSSION TITLE
Deploy on Railway - not working

### DISCUSSION THREAD

**Discussion by vishalkadam47** [Lissy93/dashy#1702] [Help and Support]
I have used this template https://railway.app/template/MtdjAQ but shows error 
```
-  Building for production...

 WARN  A new version of sass-loader is available. Please upgrade for best experience.

error Command failed with signal "SIGKILL".

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

node:internal/process/promises:289

            triggerUncaughtException(err, true /* fromPromise */);

            ^

 

RpcIpcMessagePortClosedError: Cannot send the message - the message port has been closed for the process 78.

    at /app/node_modules/@vue/cli-plugin-typescript/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/rpc-ipc/RpcIpcMessagePort.js:47:47

    at process.processTicksAndRejections (node:internal/process/task_queues:81:21) {

  code: undefined,

  signal: undefined

}

 

Node.js v20.11.1

node:internal/process/promises:289

            triggerUncaughtException(err, true /* fromPromise */);

            ^

 

RpcIpcMessagePortClosedError: Cannot send the message - the message port has been closed for the process 79.

    at /app/node_modules/@vue/cli-plugin-typescript/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/rpc-ipc/RpcIpcMessagePort.js:47:47

    at process.processTicksAndRejections (node:internal/process/task_queues:81:21) {

  code: undefined,

  signal: undefined

}

 

Node.js v20.11.1

ERROR: "build" exited with 1.

error Command failed with exit code 1.

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

container event container died
```

**Comment by FraglyG** [comment]
Template creator here, I tried reproducing and couldn't get it to throw that error, when does it error for you? Have you tried deploying this template again and deleting the previous one? 

Btw, for anyone following there's a duplicate thread for this on Railway's help forum [here](https://help.railway.app/templates/dashy-c5c1f947)

**Comment by vishalkadam47** [comment]
this is the first time i tried never deployed before

**Comment by vishalkadam47** [comment]
same issue in vercel this is the fix, but how to add build command in railway
```
build command: NODE_OPTIONS=--openssl-legacy-provider yarn build
install command: yarn install --ignore-engines
```

**Reply by FraglyG** [reply]
You can't change the build command on the Railway Dashy template because it's built from the Dashy docker image. 

I believe you could:
- fork the Dashy repo, change the build command in the `Dockerfile` and deploy on Railway through that repo
- or keep using the Dashy template but add the `NODE_OPTIONS` service variable with the value `--openssl-legacy-provider` to kind of mimic the fix you found

I can't reproduce this though so I'm not yet willing to change the template itself to accommodate this fix, but you can apply the fix yourself and see if it works

**Reply by vishalkadam47** [reply]
nope adding a variable also doesn't work, deployment is successful but when we access the link the app gets crashed
![image](https://github.com/user-attachments/assets/dd7ad0e4-ce1c-4fda-a9e5-6a424d370f18)
![image](https://github.com/user-attachments/assets/8b803908-a870-4e4d-b734-77cd53822215)
![image](https://github.com/user-attachments/assets/4b7abd6c-9986-482a-b849-ce285fbd5130)

**Reply by FraglyG** [reply]
Then the fix you found wasn't the solution, it's really strange because I cannot for the life of me reproduce this. When I deploy the template it deploys perfectly fine

My best and only guess would be that it spikes to a high memory usage at start going above 500mb (which is the max for trial on Railway) causing it to fail, I looked at previous issues with that RPC IPC error and all of them were out of memory related issues. I'm somewhat confident that it'd fix itself if you upgrade to Hobby plan on Railway
